### PR TITLE
16-add-scalafmt - added scalafmt to project

### PR DESCRIPTION
**Pull request by cricanr**
added scalafmt and formatted existing classes

### CODE REVIEW

File: app/compose/Aggregator.scala [hunk 1]
```diff
@@ -16,11 +20,11 @@ import scala.concurrent.{ExecutionContext, Future}
 class Aggregator @Inject()(movieSearchService: MovieSearchService,
                            movieInfoService: MovieInfoService,
                            artistInfoService: ArtistInfoService,
-                           ratingService: RatingService)
-                          (implicit ec: ExecutionContext,
-                           mat: Materializer,
-                           system: ActorSystem,
-                           ws: WSClient) {
+                           ratingService: RatingService)(
+    implicit ec: ExecutionContext,
```
Comment (crisanlucid) [inline app/compose/Aggregator.scala:24]: is not proper indent the code;
```
@Inject()(movieService: MovieService, 
          artistService: ArtistService
         )(
          ratingService: RatingService
         )

```

File: app/compose/Aggregator.scala [hunk 2]
```diff
@@ -32,14 +36,21 @@ class Aggregator @Inject()(movieSearchService: MovieSearchService,
...
+            artistId(moviesInfoResult))
           genresResult <- futureGenresResult
         } yield {
-          composer.compose(moviesInfoResult,
-            artistInfosResult,
-            genresResult,
-            ratingService.ratingAverageInfos(moviesInfoResult)).asJson.toString
+          composer
+            .compose(moviesInfoResult,
+                     artistInfosResult,
```
Comment (crisanlucid) [inline app/compose/Aggregator.scala:49]: JS style
```
composer.compose( moviesInfoResult,
                  artistInfosResult,
                  genresResult,
                  ratingService.ratingAverageInfos(moviesInfoResult))
        .asJson
        .toString
```

File: app/compose/Aggregator.scala [hunk 3]
```diff
@@ -49,12 +60,17 @@ class Aggregator @Inject()(movieSearchService: MovieSearchService,
 
   private def handleFailure(exception: Throwable): String = {
     exception match {
-      case failure: InvalidQueryParamException => s"Failure calling downstream service due to invalid query parameters supplied: ${failureInfo(failure)}"
-      case failure: GenericFailureCalling => s"Failure calling downstream service: ${failureInfo(failure)}"
-      case failure: FailureDecodingJson => s"Failure decoding json: ${failureInfo(failure)}"
-      case _@failure => s"Generic failure calling downstream service: ${failureInfo(failure)}"
+      case failure: InvalidQueryParamException =>
+        s"Failure calling downstream service due to invalid query parameters supplied: ${failureInfo(failure)}"
```
Comment (crisanlucid) [inline app/compose/Aggregator.scala:64]: this is not a good practice; to limit your code 80 characters. every Developer has a big screen not just the laptop; I prefer the old style everything in one line

File: app/compose/Composer.scala [hunk 4]
```diff
@@ -7,19 +7,46 @@ class Composer {
...
+      val casts = movieInfoResult.cast.flatMap { id =>
+        artistInfos.data.collect {
+          case artistInfo if artistInfo.id == id =>
+            Cast(artistInfo.id,
+                 artistInfo.gender,
+                 artistInfo.name,
+                 artistInfo.profilePath)
+        }
+      }
+      MovieDetail(
```
Comment (crisanlucid) [inline app/compose/Composer.scala:38]: `space` need from val `cast` and new variable

File: app/controllers/MoviesController.scala [hunk 5]
```diff
@@ -14,19 +14,23 @@ import scala.language.postfixOps
...
 
   def movies: Action[AnyContent] = Action.async { implicit request =>
     val maybeMoviesSearchParameters = QueryParametersHelper(request.queryString)
-    maybeMoviesSearchParameters.map(moviesSearchParams => aggregator.aggregate(moviesSearchParams).map(aggregatedResult => Ok(aggregatedResult)))
+    maybeMoviesSearchParameters
+      .map(
+        moviesSearchParams =>
+          aggregator
+            .aggregate(moviesSearchParams)
+            .map(aggregatedResult => Ok(aggregatedResult)))
```
Comment (crisanlucid) [inline app/controllers/MoviesController.scala:33]: nice

File: app/models/MovieInfo.scala [hunk 6]
```diff
@@ -1,7 +1,19 @@
...
+                     overview: String,
+                     popularity: Float,
+                     runtime: Int,
+                     releaseDate: String,
+                     revenue: Int,
+                     budget: Int,
+                     posterPath: String,
+                     originalLanguage: String,
+                     genres: Seq[Int],
+                     cast: Seq[Int])
```
Comment (crisanlucid) [inline app/models/MovieInfo.scala:18]: empty space

File: docker-compose.yml [hunk 7]
```diff
@@ -1,26 +1,26 @@
...
-    - MAX_ALLOWED=5
-  ratingsstream:
-    image: "keremk/ratingsstream:v0.3"
-    ports:
-    - "3020:3000"
\ No newline at end of file
+ movieinfo:
+   image: "keremk/movieinfo:v0.3"
+   ports:
+   - "3030:3030"
```
Comment (crisanlucid) [inline docker-compose.yml]: need tabs:  YAML files is following this model: 
```services:
    web:
        image: app:web-1.0.0
        ports:
            - "8080:80"
        links:
            - php

```

File: project/plugins.sbt [hunk 8]
```diff
@@ -4,4 +4,6 @@ resolvers += "Typesafe repository" at "http://repo.typesafe.com/typesafe/release
 
 addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.6.15")
 
-addSbtPlugin("com.tapad" % "sbt-docker-compose" % "1.0.34")
\ No newline at end of file
+addSbtPlugin("com.tapad" % "sbt-docker-compose" % "1.0.34")
+
+addSbtPlugin("com.lucidchart" % "sbt-scalafmt" % "1.15")
```
Comment (crisanlucid) [inline project/plugins.sbt:9]: for chart library; Lucian ;   lucidchart -> we can use this brand name in our famous;

File: app/compose/Aggregator.scala (hunk 1 above)
Comment (cricanr) [reply app/compose/Aggregator.scala:24]: reformatted using scalafmt class

File: app/compose/Aggregator.scala (hunk 2 above)
Comment (cricanr) [reply app/compose/Aggregator.scala:49]: reformatted using scalafmt class

File: app/compose/Aggregator.scala (hunk 3 above)
Comment (cricanr) [reply app/compose/Aggregator.scala:64]: reformatted using scalafmt class

File: app/compose/Composer.scala (hunk 4 above)
Comment (cricanr) [reply app/compose/Composer.scala:38]: reformatted using scalafmt class

File: app/models/MovieInfo.scala (hunk 6 above)
Comment (cricanr) [reply app/models/MovieInfo.scala:18]: reformatted using scalafmt class

File: docker-compose.yml (hunk 7 above)
Comment (cricanr) [reply docker-compose.yml]: reformatted

File: project/plugins.sbt (hunk 8 above)
Comment (cricanr) [reply project/plugins.sbt:9]: not sure what you mean :)

File: app/compose/Aggregator.scala (hunk 2 above)
Comment (crisanlucid) [reply app/compose/Aggregator.scala:49]: pls provide a doc for scalafmt

File: project/plugins.sbt (hunk 8 above)
Comment (crisanlucid) [reply project/plugins.sbt:9]: just an ideea; how to take advantage of faimous brand.   you can call me  Lucian from Lucien Alliance ( SG1- movie)  or Crisan Lucid, like lucidchard

**Review by crisanlucid — APPROVED** [review]

File: app/compose/Aggregator.scala (hunk 2 above)
Comment (cricanr) [reply app/compose/Aggregator.scala:49]: Ok, sure, will do now. Thanks!

Comment (cricanr) [reply app/compose/Aggregator.scala:49]: Done, will merge now.

**Pull request merged**